File: evaluate-reverse-polish-notation.cpp
#include<vector>
class Solution {
public:
    //a stack has to be used for this 
    int evalRPN(vector<string>& token) {
        vector<int>s;
        int a;
        int b;
        int res;
        for(int i=0;i<token.size();i++){
            if(token[i]=="+" || token[i]=="-" || token[i]=="/" || token[i]=="*"){
                a=s.back();
                s.pop_back();
                b=s.back();
                s.pop_back();
                if( token[i]=="+"){
                    res=a+b;
                }
                if( token[i]=="-"){
                    res=b-a;
                }
                if( token[i]=="*"){
                    res=a*b;
                }
                if( token[i]=="/"){
                    res=b/a;
                }
               // cout<<res<<a<< b<<" \n";
                s.push_back(res);
            }
            else{
                string x=token[i];
                int operand=stoi(x);
                s.push_back(operand);
            }
        }
        return s.back();
    }
};

File: evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        # Bs stack question 
        stack = []
        operators = ['/','+', '-', '*']
        for item in tokens:
            if item in operators:
                # apply operation
                a1 = stack.pop()
                a2 = stack.pop()
                res = int(eval(f"{a2}{item}{a1}"))
                stack.append(res)
            else:
                stack.append(item)
        return int(stack[0])